### PR TITLE
Added volume slider and repeat button.

### PR DESCRIPTION
**Pull request by MBlais13**
Hope this one works, ill be adding a few other features in the future so ill try to figure out how to do this pr stuff on a private project.

### CODE REVIEW

**Comment by KieranBond** [comment]
Thanks for tidying it up @MBlais13 . I'll take a look when I can - I had a brief peek at it yesterday and it looked good but I did have some thoughts.

I'd be happy to try and help explain the process, it can be quite confusing when you're new to it!

**Comment by MBlais13** [comment]
For sure. That would be super helpful if you could explain that to me, thanks!

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 1]
```diff
@@ -1,420 +1,420 @@
...
-                                return InvalidReturn;
-                        }
-                    }
-                }
-            }
-            else
-            {
-                return InvalidReturn;
-            }
+﻿using Microsoft.Win32;
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs:1]: Any idea why this whole file has seems to have been flagged as edited?

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 2]
```diff
@@ -443,6 +443,22 @@ private async void PollSongChange()
...
+            m_spotifyWebAPI.SetRepeatMode(RepeatState.Track);
+        }
+        public void RepeatSongOff() //repeat the current track
+        {
+            m_spotifyWebAPI.SetRepeatMode(RepeatState.Off);
+        }
+
+        public void VolumeChangedEvent(double sv) //changed spotify volume not application volume
+        {
+            int val = Convert.ToInt32(sv);
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs]: Nitpicky comment - Could we name this variable better?

File: MiniSpotify/MiniSpotify/Source/APIRequestor.cs [hunk 3]
```diff
@@ -443,6 +443,22 @@ private async void PollSongChange()
...
             }
+        }
+        public void RepeatSongOn() //repeat the current track
+        {
+            m_spotifyWebAPI.SetRepeatMode(RepeatState.Track);
+        }
+        public void RepeatSongOff() //repeat the current track
+        {
+            m_spotifyWebAPI.SetRepeatMode(RepeatState.Off);
+        }
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/APIRequestor.cs:454]: Isn't there 3 types of repeat? Repeat off, repeat once, repeat lots?

Would be good to have all 3 as an option. Happy to help if you're not sure how to approach this.

File: MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs [hunk 4]
```diff
@@ -41,6 +42,11 @@ public MainWindow()
             Closing += (e, f) => APIRequestor.Instance.Close();
         }
 
+        public void Slider(FullTrack a_latestTrack = null)
```
Comment (KieranBond) [inline MiniSpotify/MiniSpotify/Source/MainWindow.xaml.cs]: What's the purpose of this empty function? Was it created by accident?

**Review by KieranBond — CHANGES_REQUESTED** [review]
I've got a few thoughts below. Nothing major - Just some discussion points.

I love the idea behind the change though. 

Out of curiosity, how are you editing files? Are you using an IDE? I ask as there's a lot of formatting changes in these past two PR's; Formatting changes aren't a problem, they just make finding changes harder is all.

**Comment by KieranBond** [comment]
@MBlais13 Hey Michael. I've just made a PR (and will be merging this in) that is likely going to break this branch big time.

Once I've merged my PR in - Would you like to just upload the assets/icons you had used in this PR and then I can add the functionality to them? 

Sorry for how painful this one has been!

**Comment by MBlais13** [comment]
Is it possible to get your discord or something where we can chat about the PR easier? It seems like I've made this a bit more complicated than it should be.

**Comment by KieranBond** [comment]
Absolutely. I've not helped in making this easy so perhaps Discord will be easier. Send me a friend request - Kieran#0012 (0 0 1 2 (Github is hiding 0's to me))

**Pull request closed without merging**